cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+
+    Args:
+        meas_spec: The specification with the particular run used to
+            gather these bitstrings. There should be a 1-to-1 correspondence
+            between bitstring accumulators and circuits run on a quantum
+            sampler.
+        simul_settings: The list of settings consistent with this
+            measurement spec, usually the result of grouping a list
+            of requested settings. This list need not be exhausted:
+            any setting consistent with the `meas_spec` can be queried
+            with methods that take a setting as argument (e.g. `mean`,
+            `variance`) whether or not they are provided up-front in
+            `simul_settings`. Those methods that do *not* take a setting
+            as an argument (e.g. `means`, `variances`) will report all values
+            for the settings in `simul_settings`.
+        qubit_to_index: A mapping from qubits to contiguous indices starting
+            from zero. This allows us to store bitstrings as a 2d numpy array.
+        chunksizes: This class accumulates bitstrings from potentially several
+            "chunked" processor runs. Each chunk has a certain number of
+            repetitions, recorded in this array. This theoretically
+            allows you to re-split up the bitstring array should the need
+            arise. The total number of repetitions is the sum of this 1d array.
+        timestamps: We record a timestamp for each request/chunk. This
+            1d array will have the same length as `chunksizes`.
+    """
+
+    def __init__(
+        self,
+        meas_spec: _MeasurementSpec,
+        simul_settings: List[InitObsSetting],
+        qubit_to_index: Dict['cirq.Qid', int],
+        bitstrings: np.ndarray = None,
+        chunksizes: np.ndarray = None,
+        timestamps: np.ndarray = None,
+    ):
+        self._meas_spec = meas_spec
+        self._simul_settings = simul_settings
+        self._qubit_to_index = qubit_to_index
+
+        if bitstrings is None:
+            n_bits = len(qubit_to_index)
+            self.bitstrings = np.zeros((0, n_bits), dtype=np.uint8)
+        else:
+            self.bitstrings = np.asarray(bitstrings, dtype=np.uint8)
+
+        if chunksizes is None:
+            self.chunksizes = np.zeros((0,), dtype=np.int64)
+        else:
+            self.chunksizes = np.asarray(chunksizes, dtype=np.int64)
+
+        if timestamps is None:
+            self.timestamps = np.zeros((0,), dtype='datetime64[us]')
+        else:
+            self.timestamps = np.asarray(timestamps, dtype='datetime64[us]')
+
+        if len(self.chunksizes) != len(self.timestamps):
+            # coverage: ignore
+            raise ValueError(
+                "Invalid BitstringAccumulator state. "
+                "`chunksizes` and `timestamps` must have the same length."
+            )
+
+        if np.sum(self.chunksizes) != len(self.bitstrings):
+            # coverage: ignore
```

Comment:
Do you mind writing the test for these errors?

Reply:
done